packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart

```diff
@@ -537,14 +713,165 @@ class _ListWheelScrollViewState extends State<ListWheelScrollView> {
+
+  @override
+  void performRebuild() {
+    _childWidgets.clear();
+    super.performRebuild();
+    if (_childElements.isEmpty)
+      return;
+
+    final int firstIndex = _childElements.firstKey();
+    final int lastIndex = _childElements.lastKey();
+
+    for (int index = firstIndex; index <= lastIndex; ++index) {
+      final Element newChild = updateChild(_childElements[index], retrieveWidget(index), index);
+      if (newChild != null) {
+        _childElements[index] = newChild;
+      } else {
+        _childElements.remove(index);
+      }
+    }
+  }
+
+  /// Ask the underlying delegate for a widget at the given index.
+  ///
+  /// Normally the builder is only called once for each index and the result
+  /// will be cached. However when the element is rebuilt, the cache will be
+  /// cleared.
+  ///
+  /// This function is often called when a new child needs to be created.
+  Widget retrieveWidget(int index) {
+    return _childWidgets.putIfAbsent(index, () => widget.childDelegate.build(this, index));
+  }
+
+  @override
+  bool childExistsAt(int index) => retrieveWidget(index) != null;
+
+  @override
+  void createChild(int index, { @required RenderBox after }) {
+    owner.buildScope(this, () {
+      final bool insertFirst = after == null;
+      assert(insertFirst || _childElements[index - 1] != null);
+      Element newChild;
+      newChild = updateChild(_childElements[index], retrieveWidget(index), index);
+      if (newChild != null) {
+        _childElements[index] = newChild;
+      } else {
+        _childElements.remove(index);
+      }
+    });
+  }
+
+  @override
+  void removeChild(RenderBox child) {
+    final int index = renderObject.indexOf(child);
+    owner.buildScope(this, () {
+      assert(_childElements.containsKey(index));
+      final Element result = updateChild(_childElements[index], null, index);
+      assert(result == null);
+      _childElements.remove(index);
+      assert(!_childElements.containsKey(index));
+    });
+  }
+
+  @override
+  Element updateChild(Element child, Widget newWidget, dynamic newSlot) {
+    final ListWheelParentData oldParentData = child?.renderObject?.parentData;
+    final Element newChild = super.updateChild(child, newWidget, newSlot);
+    final ListWheelParentData newParentData = newChild?.renderObject?.parentData;
+    if (newParentData != null) {
+      newParentData.index = newSlot;
+      if (oldParentData != null)
+        newParentData.offset = oldParentData.offset;
+    }
+
+    return newChild;
+  }
+
+  @override
+  void insertChildRenderObject(RenderObject child, int slot) {
+    final RenderListWheelViewport renderObject = this.renderObject;
+    assert(renderObject.debugValidateChild(child));
+    renderObject.insert(child, after: _childElements[slot - 1]?.renderObject);
+    assert(renderObject == this.renderObject);
+  }
+
+  @override
+  void moveChildRenderObject(RenderObject child, dynamic slot) {
```

Comment:
Is it guaranteed that this will only be called to shuffle the list?

What if someone does something like this:

```
new ListWheelScrollView(
  children: const [
    Text('foo 1'),
    Text('foo 2'),
  ],
);
```

Then later they do:
```
new ListWheelScrollView(
  children: const [
    Text('foo 2'),
    Text('foo 1'),
  ],
);
```

Will that trigger this assert?

Reply:
I'm not sure.
With your example, I think the framework will try to rebuild, not moving render objects around.
This assertion is also in sliver.dart, and it makes sense to me too because we are maintaining the children in contiguous order, so that each child has a unique index, and moving them around will likely cause errors.

Reply:
OK. Be sure to add a test that this works.

Also, be sure to use either const widgets or global keys for the test, so that we can be sure it has to try to reorder the widgets.